### ISSUE TITLE
Escape reserved words used as identifier in generated sql

### ISSUE THREAD

**Issue by jopicornell** [go-jet/jet#27]
Hi there!

I love this idea, I'm building a test app with this library and I found a problem. I'm using a "user" table on PostgreSQL to store my users. This table is on a schema (image_gallery), and if I run:
```
 statement := SELECT(Picture.AllColumns).
		FROM(
			Picture.INNER_JOIN(User, Picture.UserID.EQ(User.UserID)),
		)
```
the query fails to execute. The problem is that schema is not being chained to the fields, so the query looks like this:
```
SELECT picture.picture_id AS "picture.picture_id",
     picture.user_id AS "picture.user_id",
     picture.image_id AS "picture.image_id",
     picture.title AS "picture.title",
     picture.description AS "picture.description",
     picture.created AS "picture.created",
     picture.num_likes AS "picture.num_likes",
     picture.num_comments AS "picture.num_comments"
FROM image_gallery.picture
     INNER JOIN image_gallery.user ON (picture.user_id = user.user_id); 
```
PostgreSQL complains about a wrong syntax on the "user.user_id". I guess "user" is some reserved word. I see that the solution is to scape the table names always like this:
```
SELECT "picture".picture_id AS "picture.picture_id",
       "picture".user_id AS "picture.user_id",
       "picture".image_id AS "picture.image_id",
       "picture".title AS "picture.title",
       "picture".description AS "picture.description",
       "picture".created AS "picture.created",
       "picture".num_likes AS "picture.num_likes",
       "picture".num_comments AS "picture.num_comments"
FROM image_gallery."picture"
         INNER JOIN image_gallery."user" ON ("picture".user_id = "user".user_id);
```

I think it is an easy solution and will help a lot to avoid collision with reserved words, don't you think? I think I can contribute to the project so if you see this solution as correct (I think for Mysql and MariaDB we can use ` scaping char) I can check if I can contribute to the project.

Cheers!

**Comment by go-jet** [comment]
Hi @jopicornell. Good catch. SQL reserved words should be escaped.
There is already some escape logic in place for PostgreSQL, MySQL and MariaDB. If the identifier contains non ascii or uppercase character or character which is neither number or letter or '_', those identifier will be escaped. 
https://github.com/go-jet/jet/blob/940fc06d9a2e2f146cd5b67f05dc96bb26253676/internal/jet/sql_builder.go#L100-L106

To escape reserved word, this if condition should be extended with additional check `s.Dialect.IsReservedWord(name) || ...`

`IsReservedWord` does not exist right now. Feel free to contribute if you have time.

The reason not escaping all identifiers all the time is to have better readability of generated SQL.

**Comment by jopicornell** [comment]
Great @go-jet ! I'd like for sure, but I tried to run tests and there are some errors saying:
```
../../../gotest.tools/assert/assert.go:74:2: code in directory /home/jopicornell/go/src/gotest.tools/assert/cmp expects import "gotest.tools/assert/cmp"
../../../gotest.tools/assert/assert.go:75:2: code in directory /home/jopicornell/go/src/gotest.tools/internal/format expects import "gotest.tools/internal/format"
../../../gotest.tools/assert/assert.go:76:2: code in directory /home/jopicornell/go/src/gotest.tools/internal/source expects import "gotest.tools/internal/source"
```

Do you have any discord or slack so we can communicate faster and not flood this issue? If not, I'm sorry posting here questions regarding contributing :slightly_smiling_face: 

I tried searching for this errors and seems like gotest.tools is requiring imports in some way, but I don't know if I'm doing something wrong to run the tests.

**Comment by go-jet** [comment]
@jopicornell If you are on the Mac, and the test that fails is `TestCmdGenerator` you can ignore it.
I've add [gitter](https://gitter.im/go-jet/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge) link to main Readme.

**Comment by go-jet** [comment]
Hi @jopicornell. Any luck with this issue?

**Comment by jopicornell** [comment]
Hi @go-jet I manage to fix it quickly for my needs, but the problem I encounter to contribute is how to find a curated list of restricted words in sql and which method is the best to find them in that list. Also how to import into the library all of the words. I used a map with boolean, which I think is the fastest way, but a bit harder to build the list. Any suggestion?

**Comment by go-jet** [comment]
Postgresql: https://www.postgresql.org/docs/12/sql-keywords-appendix.html
MySQL: https://dev.mysql.com/doc/refman/8.0/en/keywords.html
Map with boolean seems right. You can construct a slice of keywords and build a map from that slice inside dialect constructor.

**Comment by go-jet** [comment]
`Escaping reserved words used as identifier in generated SQL` is fixed with the release [v2.2.0](https://github.com/go-jet/jet/releases/tag/v2.2.0).

@jopicornell The bug was open for long time and it had a medium severity, so It had to go with a new release. Feel free to contribute in the feature, and report any issue you find.

Cheers!
